network/p2p/observability.go

```diff
@@ -0,0 +1,112 @@
+package p2pv1
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/ssvlabs/ssv/network/peers"
+	"github.com/ssvlabs/ssv/network/topics"
+	"github.com/ssvlabs/ssv/observability"
+)
+
+const (
+	observabilityName      = "github.com/ssvlabs/ssv/network/p2p"
+	observabilityNamespace = "ssv.p2p"
+)
+
+var (
+	meter = otel.Meter(observabilityName)
+
+	peersConnectedGauge = observability.NewMetric(
+		meter.Int64Gauge(
+			metricName("peers.connected"),
+			metric.WithUnit("{peer}"),
+			metric.WithDescription("number of connected peers")))
+
+	connectionsGauge = observability.NewMetric(
+		meter.Int64Gauge(
+			metricName("connections.active"),
+			metric.WithUnit("{connection}"),
+			metric.WithDescription("number of active connections")))
```

Comment:
I don't get the difference between these two metrics?
shouldn't it be the same?

Reply:
Sure.
If to look into implementation - `peer` and `connection` are two distinctly different entities, although they relate to each other, and their relationships are one to many, meaning one `peer` has one or more `connections`. `Connection` also has a `direction` attribute - either `inbound` or `outbound`. 
[Here](https://github.com/ssvlabs/ssv/blob/otel-metrics/network/p2p/observability.go#L63) it is visible how connections can be fetched via providing the Peer ID, and it is a collection of items. We separate them on our dashboard. Frankly, in most of the cases, it's 1:1, although since API tells us it's a collection, I assume it is possible it to be more than connection per peer. 
Wdyt?

Reply:
It seems to be possible in terms of networking and libp2p, but AFAIK we limit it on our end.
So peers.connected will always be 1:1 with connections just without the direction directive. maybe we can consolidate them. @zktaiga @moshe-blox wdyt?